pkg/core/state/client.go

```diff
@@ -32,22 +35,45 @@ func NewClient(db execution.QueryExecer, logger hclog.Logger) *Client {
 	}
 }
 
-func (c *Client) MigrateCore(ctx context.Context, de database.DialectExecutor) error {
-	err := c.db.Exec(ctx, "CREATE SCHEMA IF NOT EXISTS cloudquery")
-	if err != nil {
-		return err
+func NewMigratedClient(ctx context.Context, dsn string, logger hclog.Logger) (*Client, error) {
```

Comment:
nit: NewClient is enough imo

Reply:
I don't want to init db and call MigrateCore every time I want an updated core. I'll fix this when I circle back to fix storage / make it transactional etc.